pandas/tests/indexes/period/methods/test_asfreq.py

```diff
@@ -128,3 +129,10 @@ def test_astype_asfreq(self):
         exp = PeriodIndex(["2011-01", "2011-02", "2011-03"], freq="3M")
         tm.assert_index_equal(pi1.asfreq("3M"), exp)
         tm.assert_index_equal(pi1.astype("period[3M]"), exp)
+
+    def test_asfreq_with_periodindex(self):
```

Comment:
I think all tests here check `asfreq` with `periodindex` 😉 perhaps `test_asfreq_with_different_n`?
```suggestion
    def test_asfreq_with_different_n(self):
```